### PR TITLE
Min/max with NaNs and imag inequalities

### LINKED ISSUE #13560: nans do not propagate through a reduction
### Summary of Problem
`nan`'s do not appear to propagate through a reduction.

### Steps to Reproduce

**Source Code:**

```chapel
var a = [1.0e150,2.0e309];
var b = [2.0e150,2.1e309];
writeln(max reduce abs(a-b));
writeln(abs(a-b));
```

produces
```
1e+150
1e+150 nan
```
whereas I would expect this to produce a `nan`.

This is on Chapel 1.19 (running on tio.run).

### PR DESCRIPTION
**Pull request by vasslitvinov**
Resolves #13560 :

* `min()` and `max()` on real numbers now propagate NaN.
* ditto `min`, `max`, `minloc`, `maxloc` reductions
* Inequality comparisons on `imag` numbers are now deprecated.

While there, revert an unintentional change made a long time ago from
`max(an int, a real cast to int)` to `max(an int, a real) cast to int`.
This allows the `Math` module to continue being dead-module-eliminated.
Without this revert, each compilation would be using `isnan()` from `Math`.

Corresponding updates to the spec and chpldocs.

Test changes include:

* Replace `test/types/complex/diten/imag_compare.chpl` with
  `test/deprecated/imag-compare.chpl`.

* Merge `sortRealImag.chpl` and `sortRealImagTuples.chpl` into `sortReal.chpl`
  in `test/library/packages/Sort/correctness/`, after removing `imag` sorting.

Testing: linux64 -futures, (earlier version) gasnet multilocale tests, numa.

### CODE REVIEW

**Comment by vasslitvinov** [comment]
@mppf was there a reason that
`test/library/packages/Sort/correctness/sortRealImagTuples.chpl`
was added in #12489 even though
`test/library/packages/Sort/correctness/sortRealImag.chpl`
added in #12452 contains tuples?

I am merging these two tests into `sortReal.chpl` in this pr.

File: test/types/real/nan-minmax.chpl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+writeln("done");
+
+// min/max reductions on ARR with 1 or 2 NANs added to it
+proc testNans() {
+  for i1 in 1..n do
+    for i2 in 1..n do
+      testNan(ARR, i1, i2);
+}
+proc testNan(in ARR, idx1, idx2) {
```
Comment (ben-albrecht) [inline test/types/real/nan-minmax.chpl]: The overloaded usage of the `ARR` identifier for global/local scope makes me a little nervous, but this looks correct to me.

**Comment by ben-albrecht** [comment]
After thought: Is there some kind of documentation (spec or primer) that should be updated to reflect that imaginary comparisons will not be permitted?

**Comment by vasslitvinov** [comment]
@ben-albrecht - I am done updating to your feedback. Let me know how it looks to you now.

**Pull request merged**